cmd/tuf/import_key.go

```diff
@@ -0,0 +1,68 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/flynn/go-docopt"
+	"github.com/theupdateframework/go-tuf"
+	"github.com/theupdateframework/go-tuf/data"
+	"github.com/theupdateframework/go-tuf/pkg/keys"
+)
+
+func init() {
+	register("import-key", cmdImportKey, `
+usage: tuf import-key [--expires=<days>] <role> <file>
+
+Import an existing signing key for the given role.
```

Comment:
What format(s)?

Reply:
It looks like right now this is "the exact format from the keys DB for another go-tuf repo".  I think that's a lot less useful than e.g. PKCS 8 keys. Also, I think we want to be able to import both public/private keys.

Reply:
This is doing both: importing a key into the key database, and adding it to a role. But often we're going to only want to do one or the other.

Reply:
So perhaps some more context on why I'm trying to add this would help answer this and the comment below. The specific use case is for adding a private key. The way I envisage this working is having a separate tool like `tuf-yubikey` that would do something like:
```
$ tuf-yubikey generate-key --serial=12345678 --slot=9c --output=yubi-snapshot.json
<prompt for management key>
New key written to yubi-snapshot.json
```
(this would be a bit more interactive to make it easier to derive slot and serial numbers)

This would generate a file like:
```
{
	"keytype": "yubi-rsa",
	"scheme": "rsassa-pss-sha256",
	"keyid_hash_algorithms": [
		"sha256",
		"sha512"
	],
	"keyval": {
		"serial": "12345678",
		"slot": "9c",
		"public": "...."
	}
}
```
But we now need a way to import this into the keystore (adding it to the private key store and then adding it to the root metadata). So you'd now just be able to do `tuf import --role=snapshot yubi-snapshot.json.

As far as I'm aware, unlike public keys the private key format is not part of the TUF specification and are effectively an implementation detail of go-tuf (but perhaps inherited from python-tuf??). Could certainly implement something like pkcs8 fairly easily, but think this is for solving a different problem (this is building up to support external key management, so you won't be able to export the private key).

Reply:
> The way I envisage this working is having a separate tool like `tuf-yubikey`

Hmm, my preference is to build as few wrappers as possible for external tools, and instead teach go-tuf about specific formats to import.

> But we now need a way to import this into the keystore (adding it to the private key store and then adding it to the root metadata). So you'd now just be able to do `tuf import --role=snapshot yubi-snapshot.json`.

Perhaps we want a more "plumbing and porcelain" approach here: we provide separate commands for manipulating the keys database and associating keys with roles. Then, you'd be able to write an ugly shell script for generating a key pair and attaching it to a role. After that, you could turn that shell script into its own command so we can do this process in one step.